test-suite/src/test/scala/org/scalajs/testsuite/javalib/LinkedListTest.scala

```diff
@@ -0,0 +1,226 @@
+      expect(ll.indexOf("three")).toEqual(2)
+      expect(ll.lastIndexOf("three")).toEqual(2)
+    }
+
+    it("should remove occurrencies of provided elements") {
+      val l = asJavaCollection(Seq("one", "two", "three", "two", "one"))
+      val ll = new LinkedList[String](l)
+
+      expect(ll.removeFirstOccurrence("one")).toBeTruthy
+      expect(ll.indexOf("one")).toEqual(3)
+      expect(ll.removeLastOccurrence("two")).toBeTruthy
+      expect(ll.lastIndexOf("two")).toEqual(0)
+      expect(ll.removeFirstOccurrence("one")).toBeTruthy
+      expect(ll.removeLastOccurrence("two")).toBeTruthy
+      expect(ll.removeFirstOccurrence("three")).toBeTruthy
+      expect(ll.removeLastOccurrence("three")).toBeFalsy
+      expect(ll.isEmpty).toBeTruthy
+    }
+
+    it("should iterate over elements in both directions") {
+      val s = Seq("one", "two", "three")
+      val l = asJavaCollection(s)
+      val ll = new LinkedList[String](l)
+
+      val iter = ll.iterator()
+      for (i <- 0 until l.size()) {
+        expect(iter.hasNext()).toBeTruthy
+        expect(iter.next()).toEqual(s(i))
+      }
+      expect(iter.hasNext()).toBeFalsy
+
+      val diter = ll.descendingIterator()
+      for (i <- (0 until l.size()).reverse) {
+        expect(diter.hasNext()).toBeTruthy
+        expect(diter.next()).toEqual(s(i))
+      }
+      expect(diter.hasNext()).toBeFalsy
+    }
+
+    it("should iterate and modify elements with a listIterator") {
+      val s = Seq("one", "two", "three")
+      val l = asJavaCollection(s)
+      val ll = new LinkedList[String](l)
+
+      val iter = ll.listIterator(1)
+
+      expect(iter.hasNext()).toBeTruthy
+      expect(iter.hasPrevious()).toBeTruthy
+
+      expect(iter.previous()).toEqual("one")
+
+      expect(iter.hasNext()).toBeTruthy
+      expect(iter.hasPrevious()).toBeFalsy
+
+      expect(iter.next()).toEqual("one")
+
+      expect(iter.next()).toEqual("two")
+      expect(iter.next()).toEqual("three")
+
+      expect(iter.hasNext()).toBeFalsy
+      expect(iter.hasPrevious()).toBeTruthy
+
+      iter.add("four")
+
+      expect(iter.hasNext()).toBeFalsy
+      expect(iter.hasPrevious()).toBeTruthy
+
+      expect(iter.previous()).toEqual("four")
+
+      iter.remove()
+
+      expect(iter.hasNext()).toBeFalsy
+      expect(iter.hasPrevious()).toBeTruthy
+      expect(iter.previous()).toEqual("three")
+      iter.set("THREE")
+      expect(iter.previous()).toEqual("two")
+      iter.set("TWO")
+      expect(iter.previous()).toEqual("one")
+      iter.set("ONE")
+      expect(iter.hasNext()).toBeTruthy
+      expect(iter.hasPrevious()).toBeFalsy
+
+      expect(iter.next()).toEqual("ONE")
+      iter.remove()
+      expect(iter.next()).toEqual("TWO")
+      iter.remove()
+      expect(iter.next()).toEqual("THREE")
+      iter.remove()
+
+      expect(iter.hasNext()).toBeFalsy
+      expect(iter.hasPrevious()).toBeFalsy
+
+      expect(ll.isEmpty()).toBeTruthy
+    }
```

Comment:
This great test should be moved to `ListTest`, so that `ArrayListTest` also benefits from it.